pkg/csi/service/wcp/controller.go

```diff
@@ -1636,24 +1639,32 @@ func (c *controller) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshot
 				return nil, logger.LogNewErrorCodef(log, codes.Internal,
 					"failed to retrieve cnsVolumeInfo for volume: %s Error: %+v", volumeID, err)
 			}
-			snapshotID, snapshotCreateTimePtr, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
+			snapshotID, cnsSnapshotInfo, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
 				volumeID, req.Name, &cnsvolume.CreateSnapshotExtraParams{
 					StorageClassName:           cnsVolumeInfo.Spec.StorageClassName,
 					StoragePolicyID:            cnsVolumeInfo.Spec.StoragePolicyID,
 					Namespace:                  cnsVolumeInfo.Spec.Namespace,
 					Capacity:                   cnsVolumeInfo.Spec.Capacity,
 					IsStorageQuotaM2FSSEnabled: isStorageQuotaM2FSSEnabled,
 				})
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to create snapshot on volume %q: %v", volumeID, err)
+			}
+			err = c.UpdateCNSVolumeInfo(ctx, cnsSnapshotInfo)
```

Comment:
There should be logic here to determine if you must update cnsvolumeinfo, basically check if the current cnsvolumeinfo.SnapshotLatestOperationCompleteTime < cnsSnapshotInfo.SnapshotOperationTimestamp, if so, then update the CnsVolumeInfo, if not, log it and return success.